plugins/techdocs/src/reader/transformers/scrollIntoNavigation.test.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { scrollIntoNavigation } from '.';
+
+jest.useFakeTimers();
+
+describe('scrollIntoNavigation', () => {
+  const transformer = scrollIntoNavigation();
+  const dom = { querySelectorAll: jest.fn().mockReturnValue([]) };
```

Comment:
There is a `createTestShadowDow` utility to help you mock the dom and write tests, here is an example:
https://github.com/lunarway/backstage/blob/feature/techdocs-scroll/plugins/techdocs/src/reader/transformers/copyToClipboard.test.ts#L32

Reply:
Nice. I took reference in the scrollToAnchor tests but this looks way better. Maybe I'll update the other tests as well to align them 👍

Reply:
I have updated the tests to use the shadowDom. Way better 🎉 Thanks for the pointers.